discord/sink.py

```diff
@@ -0,0 +1,215 @@
+    'time': 0,
+    'users': [],
+    'max_size': 0,
+}
+
+
+class Filters:
+    # TODO: Filter for max size per file; audio can be split into multiple files
+    def __init__(self, **kwargs):
+        self.filtered_users = kwargs.get('users', default_filters['users'])
+        self.seconds = kwargs.get('time', default_filters['time'])
+        self.max_size = kwargs.get('max_size', default_filters['max_size'])
+        self.finished = False
+
+    @staticmethod
+    def filter_decorator(func):  # Contains all filters
+        def _filter(self, data, user):
+            if not self.filtered_users or user in self.filtered_users:
+                return func(self, data, user)
+        return _filter
+
+    def init(self):
+        if self.seconds != 0:
+            thread = threading.Thread(target=self.wait_and_stop)
+            thread.start()
+
+    def wait_and_stop(self):
+        time.sleep(self.seconds)
+        if self.finished:
+            return
+        self.vc.stop_recording()
+
+
+class RawData:
+    """Handles raw data from Discord so that it can be decrypted and decoded to be used."""
+    def __init__(self, data, client):
+        self.data = bytearray(data)
+        self.client = client
+
+        self.header = data[:12]
+        self.data = self.data[12:]
+
+        unpacker = struct.Struct('>xxHII')
+        self.sequence, self.timestamp, self.ssrc = unpacker.unpack_from(self.header)
+        self.decrypted_data = getattr(self.client, '_decrypt_' + self.client.mode)(self.header, self.data)
+        self.decoded_data = None
+
+        self.user_id = None
+
+
+class AudioData:
+    """Handles data that's been completely decrypted and decoded and is ready to be saved to file."""
+    def __init__(self, file):
+        self.file = open(file, 'ab')
+        self.dir_path = os.path.split(file)[0]
+
+        self.finished = False
+
+    def write(self, data):
+        if self.finished:
+            raise ClientException("This AudioData is already finished writing.")
+        self.file.write(data)
+
+    def cleanup(self):
+        if self.finished:
+            raise ClientException("This AudioData is already finished writing.")
+        self.file.close()
+        self.file = os.path.join(self.dir_path, self.file.name)
+        self.finished = True
+
+    def on_format(self, encoding):
+        if not self.finished:
+            raise ClientException("This AudioData is still writing.")
+        name = os.path.split(self.file)[1]
+        name = name.split('.')[0] + f'.{encoding}'
+        self.file = os.path.join(self.dir_path, name)
+
+
+class Sink(Filters):
+    """A Sink "stores" all the audio data.
+
+        Parameters
+        ----------
+        encoding: :class:`string`
+            Valid types include wave
+        output_path: :class:`string`
+            A path to where the audio files should be output
+
+        Raises
+        ------
+        ClientException
+            That's not a valid encoding type.

```

Comment:
```suggestion
        Parameters
        ----------
        encoding: :class:`string`
            The encoding to use. Valid types include wave.
        output_path: :class:`string`
            A path to where the audio files should be output.

        Raises
        ------
        ClientException
            An invalid encoding type was specified.
```
For consistency the descriptions should end with a full stop. Changed the error description to fit more with the style of the rest of the docs.

Reply:
Thanks, I also noticed that I need to update the doc string since wave isn't the only valid encoding type anymore lol